File: MirrorMirror/MirrorMirror/Controllers/SLRootNavigationContainerViewController.m
//
//  SLRootNavigationContainerViewController.m
//  MirrorMirror
//
//  Created by Bradley Griffith on 5/31/14.
//  Copyright (c) 2014 Summer League. All rights reserved.
//

#import "SLRootNavigationContainerViewController.h"
#import "SLSlideInContainerViewController.h"
#import "SLCameraViewController.h"
#import "SLFeedViewController.h"

@interface SLRootNavigationContainerViewController ()
- (IBAction)showCamera:(id)sender;
- (IBAction)cancelCamera:(id)sender;

@property (weak, nonatomic) IBOutlet UIView *rootContainerView;
@property (nonatomic, strong) SLCameraViewController *cameraViewController;
@property (nonatomic, strong) SLFeedViewController *feedViewController;
@property (nonatomic, strong) SLSlideInContainerViewController *slideInController;

@end

@implementation SLRootNavigationContainerViewController


-(id)initWithCoder:(NSCoder *)aDecoder
{
	self = [super initWithCoder:aDecoder];
	if (self) {
		_cameraViewController = [[SLCameraViewController alloc] init];
		_feedViewController = [[SLFeedViewController alloc] init];
	}
	return self;
}

- (void)cancelCamera:(id)sender {
	[_slideInController pushViewController:_feedViewController];
}

- (IBAction)showCamera:(id)sender {
	[_slideInController popViewController];
}

- (void)setupSlideInController
{
	 _slideInController = [[SLSlideInContainerViewController alloc] initWithRootViewController:_cameraViewController];
	
	[_slideInController willMoveToParentViewController:self];
	[self addChildViewController:_slideInController];
	
	_slideInController.view.frame = _rootContainerView.bounds;
	
	[_rootContainerView addSubview:_slideInController.view];
	[_slideInController didMoveToParentViewController:self];
}

- (void)viewDidLoad
{
	[super viewDidLoad];

	[self setupSlideInController];
	[_slideInController pushViewController:_feedViewController];
}

@end

File: MirrorMirror/MirrorMirror/Controllers/SLRootNavigationContainerViewController.swift
//
//  SLRootNavigationContainerViewController.swift
//  MirrorMirror
//
//  Created by Bradley Griffith on 6/3/14.
//  Copyright (c) 2014 Summer League. All rights reserved.
//

import UIKit

class SLRootNavigationContainerViewController: UIViewController {
	
	var slideInController: SLSlideInContainerViewController! = nil
	var cameraViewController: SLCameraViewController!
	var feedViewController: SLFeedViewController!
	@IBOutlet var rootContainerView: UIView!
	
	init(coder aDecoder: NSCoder!) {
		super.init(coder:aDecoder)
		
		self.cameraViewController = SLCameraViewController()
		self.feedViewController = SLFeedViewController()
	}
	
	override func viewDidLoad() {
		self.setupSlideInController()
		self.slideInController.pushViewController(self.feedViewController)
	}
	
	func setupSlideInController() {
		self.slideInController = SLSlideInContainerViewController(rootViewController: self.cameraViewController)
		
		self.slideInController.willMoveToParentViewController(self)
		self.addChildViewController(self.slideInController)
		
		self.slideInController.view.frame = self.rootContainerView.bounds
		
		self.rootContainerView.addSubview(self.slideInController.view)
		self.slideInController.didMoveToParentViewController(self)
	}
	
	@IBAction func showCamera(AnyObject) {
		self.slideInController.popViewController()
		
		let hideCameraButton: UIBarButtonItem = UIBarButtonItem(title: "x", style: .Plain, target: self, action: "hideCamera:")
		self.navigationItem.leftBarButtonItem = hideCameraButton;
		
		//self.navigationController.navigationBar.barTintColor = UIColor.redColor()
	}
	
	func hideCamera(AnyObject) {
		self.slideInController.pushViewController(self.feedViewController)
		
		self.navigationItem.leftBarButtonItem = nil;
	}

}
